rust_src/src/data.rs

```diff
@@ -331,6 +331,20 @@ pub fn subr_name(subr: LispSubrRef) -> LispObject {
     unsafe { build_string(name) }
 }
 
+/// Return the byteorder for the machine.
+/// Returns 66 (ASCII uppercase B) for big endian machines or 108
+/// (ASCII lowercase l) for small endian machines.
+#[lisp_fn]
+pub fn byteorder() -> u8 {
```

Comment:
We can do something like this in Rust.

```rust
#[cfg(target_endian="big")]
pub fn byteorder() -> u8 {
  return 'B' as u8
}

#[cfg(target_endian="little")]
pub fn byteorder() -> u8 {
  return 'l' as u8
}
```

Reply:
Very nice.

Reply:
Which one gets the docstring?

Reply:
Is there a way to match on `target_endian` at compile time? Some char is getting returned as a u8, and it will be 'B' if it's "big" and 'l' if it's "little", and all that could potentially be figured out by the compiler. That would avoid having two function definitions. (But having two is better than doing any computation at run time.)

Reply:
```rust
#[cfg(target_endian="big")]
const BYTE_ORDER: u8 = b'B';
#[cfg(target_endian="little")]
const BYTE_ORDER: u8 = b'l';

pub fn byteorder() -> u8 {
    return BYTE_ORDER;
}
```
@nickdrozd An Alternative. However not much better, I think.

Reply:
Maybe not much better, but I think this is the way we've done it so far.

Reply:
How about
```
pub fn byteorder() -> u8{
    if cfg!(endian = "big") {
        b'B'
    } else {
        b'l'
    }
}
```
`cfg!(endian = "big")` should get replaced with a boolean at compile time, at which point the expression will be `if bool {b'B'} else {b'l'}`, and that expression should get reduced by the compiler as well.

Reply:
LGTM if the code can be compiled.

Reply:
@nickdrozd that is an option. Otherwise, the defun could call a helper function so the docstring can live on the defun.